sw/device/silicon_creator/lib/sigverify/spx_verify.c

```diff
@@ -32,26 +34,79 @@ static uint32_t sigverify_spx_verify_enabled(lifecycle_state_t lc_state) {
   }
 }
 
-rom_error_t sigverify_spx_verify(const sigverify_spx_signature_t *signature,
-                                 const sigverify_spx_key_t *key,
-                                 lifecycle_state_t lc_state,
-                                 uint32_t *flash_exec) {
+/**
+ * Shares for producing the `flash_exec_spx` value in `sigverify_spx_verify()`
+ * when SPHINCS+ signature verification is enabled. First 3 shares are generated
+ * using the `sparse-fsm-encode` script while the last share is
+ * `kSigverifySpxSuccess ^ kSpxShares[0] ^ ... ^ kSpxShares[2]` so that xor'ing
+ * all shares produces `kSigverifySpxSuccess`.
+ *
+ * Encoding generated with
+ * $ ./util/design/sparse-fsm-encode.py -d 5 -m 3 -n 32 \
+ *     -s 327911352 --language=c
+ *
+ * Minimum Hamming distance: 15
+ * Maximum Hamming distance: 19
+ * Minimum Hamming weight: 10
+ * Maximum Hamming weight: 17
+ */
+static const uint32_t kSpxVerifyShares[kSigverifySpxRootNumWords] = {
+    0xa5bda1e8,
+    0x229044e4,
+    0x94eadad8,
+    0x9eabb3c3,
+};
+
+rom_error_t sigverify_spx_verify(
+    const sigverify_spx_signature_t *signature, const sigverify_spx_key_t *key,
+    lifecycle_state_t lc_state, const void *msg_prefix_1,
+    size_t msg_prefix_1_len, const void *msg_prefix_2, size_t msg_prefix_2_len,
+    const void *msg, size_t msg_len, uint32_t *flash_exec) {
   uint32_t spx_en = launder32(sigverify_spx_verify_enabled(lc_state));
   rom_error_t error = kErrorSigverifyBadSpxSignature;
   if (launder32(spx_en) != kSigverifySpxDisabledOtp) {
-    // FIXME: Implement this
+    sigverify_spx_root_t expected_root;
+    spx_public_key_root((const uint8_t *)key, (uint32_t *)&expected_root);
```

Comment:
nit: for technical strict-aliasing compliance cast to `unsigned char` here instead of `uint8_t` (same goes for the casts in `spx_verify(...)` a few lines below). Also, I think you can avoid the cast of `expected_root` by referencing the internal field (and same for `actual_root` below).
```suggestion
    spx_public_key_root((const unsigned char *)key, expected_root.data);
```

Reply:
Good points! Apologies for not mentioning this in the original description but I was planning to revisit the interfaces that I messed up by adding 4 additional parameters which brings the total to 11 :)

I'll crate an issue from your comment so that I don't forget about this.